test/integration/parachain/staging/sequential/replace.test.ts

```diff
@@ -86,27 +86,29 @@ describe("replace", () => {
         });
 
         it("should request vault replacement", async () => {
-            const APPROX_TEN_BLOCKS_MS = 10 * 12 * 1000;
+            const APPROX_FIFTEEN_BLOCKS_MS = 15 * 12 * 1000;
             for (const vault_3_id of vault_3_ids) {
                 // try to set value above dust + estimated fees
                 const issueAmount = dustValue.add(feesEstimate).mul(1.2);
                 const replaceAmount = dustValue;
                 await issueSingle(interBtcAPI, bitcoinCoreClient, userAccount, issueAmount, vault_3_id);
 
-                interBtcAPI.setAccount(vault_3);
                 const collateralCurrency = await currencyIdToMonetaryCurrency(
                     assetRegistry,
                     vault_3_id.currencies.collateral
                 );
-                const [foundEvent] = await Promise.all([
-                    waitForEvent(interBtcAPI, api.events.replace.AcceptReplace, true, APPROX_TEN_BLOCKS_MS),
-                    interBtcAPI.replace.request(replaceAmount, collateralCurrency),
-                ]);
-
-                interBtcAPI.setAccount(userAccount);
+                const foundEventPromise = waitForEvent(
+                    interBtcAPI,
+                    api.events.replace.AcceptReplace,
+                    true,
+                    APPROX_FIFTEEN_BLOCKS_MS
+                );
+                await callWith(interBtcAPI, vault_3, async () =>
+                    interBtcAPI.replace.request(replaceAmount, collateralCurrency)
+                );
 
-                assert.isTrue(
-                    foundEvent,
+                await expect(foundEventPromise).to.eventually.be.equal(
```

Comment:
This is nice, I wonder though if the change doesn't reintroduce the race condition you tried to avoid when previously you batched the Promises in `Promise.all`. You can still do
```TypeScript
await Promise.all([
    callWith(interBtcAPI, vault_3, async () =>
        interBtcAPI.replace.request(replaceAmount, collateralCurrency)
    ),
    expect(foundEventPromise).to.eventually.be.equal( ... )
]);
```